client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js

```diff
@@ -0,0 +1,152 @@
+	let store;
+
+	useFakeTimers( fakeClock => clock = fakeClock );
+
+	beforeEach( () => {
+		dispatch = spy();
+		store = { dispatch };
+	} );
+
+	it( 'should pass through initially successful requests', () => {
+		const inbound = { nextData: 1, originalRequest: getSites, store };
+
+		expect( retryOnFailure( inbound ) ).to.equal( inbound );
+
+		clock.tick( 20000 );
+		expect( dispatch ).to.have.not.been.called;
+	} );
+
+	it( 'should pass through no-retry failed requests', () => {
+		const originalRequest = { ...getSites, options: { retryPolicy: noRetry() } };
+		const inbound = { nextError, originalRequest, store };
+
+		expect( retryOnFailure( inbound ) ).to.equal( inbound );
+
+		clock.tick( 20000 );
+		expect( dispatch ).to.have.not.been.called;
+	} );
+
+	it( 'should pass through POST requests', () => {
+		const originalRequest = { ...getSites, method: 'POST' };
+		const inbound = { nextError, originalRequest, store };
+
+		expect( retryOnFailure( inbound ) ).to.equal( inbound );
+
+		clock.tick( 20000 );
+		expect( dispatch ).to.have.not.been.called;
+	} );
+
+	it( 'should requeue a plain failed request', () => {
+		const inbound = { nextError, originalRequest: getSites, store };
+
+		expect( retryWithDelay( 1337 )( inbound ) ).to.have.property( 'shouldAbort', true );
+		expect( dispatch ).to.have.not.been.called;
+
+		clock.tick( 1337 );
+		expect( dispatch ).to.have.been.calledWith( withRetries( 1 )( getSites ) );
+	} );
+
+	it( 'should requeue only up to `maxAttempts`', () => {
+		const originalRequest = { ...getSites, options: { retryPolicy: { maxAttempts: 3 } } };
+		const inbound = { nextError, originalRequest, store };
+		const retryIt = retryWithDelay( 1337 );
+
+		// retry 1
+		expect( retryIt( inbound ) ).to.have.property( 'shouldAbort', true );
+		expect( dispatch ).to.have.not.been.called;
+
+		clock.tick( 1337 );
+		expect( dispatch ).to.have.been.calledWith( withRetries( 1 )( originalRequest ) );
+
+		// retry 2
+		expect( retryIt( { ...inbound, originalRequest: dispatch.lastCall.args[ 0 ] } ) ).to.have.property( 'shouldAbort', true );
+		expect( dispatch.callCount ).to.equal( 1 );
+
+		clock.tick( 1337 );
+		expect( dispatch.callCount ).to.equal( 2 );
+		expect( dispatch ).to.have.been.calledWith( withRetries( 2 )( originalRequest ) );
+
+		// retry 3
+		expect( retryIt( { ...inbound, originalRequest: dispatch.lastCall.args[ 0 ] } ) ).to.have.property( 'shouldAbort', true );
+		expect( dispatch.callCount ).to.equal( 2 );
+
+		clock.tick( 1337 );
+		expect( dispatch.callCount ).to.equal( 3 );
+		expect( dispatch ).to.have.been.calledWith( withRetries( 3 )( originalRequest ) );
+
+		// retry 4
+		const finalRequest = { ...inbound, originalRequest: dispatch.lastCall.args[ 0 ] };
+		expect( retryIt( finalRequest ) ).to.equal( finalRequest );
+		expect( dispatch.callCount ).to.equal( 3 );
+
+		clock.tick( 1337 );
+		expect( dispatch.callCount ).to.equal( 3 );
+	} );
+
+	it( 'should handle `exponentialBackoff`', () => {
```

Comment:
It feels to me like each retry case should be its own unit test.  

each case is independent of the cases before it right?

I also wouldn't block 🚢 for this